src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,180 @@
+      const chatsToUpdate = privateChats.filter(
+        (chat) => !chat.isRead && chat.from === sender.id
+      );
+      chatsToUpdate.map(
+        (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+      );
+      setSelectedUser(sender);
+    },
+    [privateChats, user]
+  );
+
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = useCallback(
+    async (data: { messageToTheBand: string }) => {
+      chatContext &&
+        user &&
+        chatContext.sendPrivateChat(
+          user.uid,
+          selectedUser!.id,
+          data.messageToTheBand
+        );
+    },
+    [chatContext, selectedUser, user]
+  );
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  const discussions = useMemo(() => {
+    return Object.keys(discussionPartnerWithLastMessageExchanged).sort((a, b) =>
+      discussionPartnerWithLastMessageExchanged[b].ts_utc
+        .valueOf()
+        .localeCompare(
+          discussionPartnerWithLastMessageExchanged[a].ts_utc.valueOf()
+        )
+    );
+  }, [discussionPartnerWithLastMessageExchanged]);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
+      />
+      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
+    </>
+  ) : (
+    <>
+      {hasPrivateChats && (
+        <div className="private-container show">
+          <div className="private-messages-list">
+            {discussions.map((userId: string) => {
+              const sender = { ...chatUsers![userId], id: userId };
+              const lastMessageExchanged =
+                discussionPartnerWithLastMessageExchanged?.[userId];
+              const profileName = sender.anonMode
+                ? DEFAULT_PARTY_NAME
+                : sender.partyName;
+
+              return (
+                <div
+                  key={userId}
+                  className="private-message-item"
+                  onClick={() => onClickOnSender(sender)}
+                  id="private-chat-modal-select-private-recipient"
+                >
+                  <UserProfilePicture
+                    avatarClassName="private-message-author-pic"
+                    user={sender}
+                    setSelectedUserProfile={() => null}
```

Comment:
This will cause re-renders due to the function reference.

To fix it..

Outside of the component:

```typescript
const noopHandler = () => {}
```

Then here you can do:

```suggestion
                    setSelectedUserProfile={noopHandler}
```

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044